### PR TITLE
add max_bins_percent for histogram

### PR DESCRIPTION
**Pull request by chesterxgchen**

### CODE REVIEW

File: nvflare/app_common/executors/statistics/statistics_executor.py [hunk 1]
```diff
@@ -227,10 +229,10 @@ def get_histogram(
         num_of_bins: int = self.get_number_of_bins(feature_name, hist_config)
         bin_range: List[int] = self.get_bin_range(feature_name, global_min_value, global_max_value, hist_config)
         item_count = self.stats_generator.count(dataset_name, feature_name)
-        if num_of_bins >= item_count:
+        if num_of_bins >= item_count * self.max_bins_percent:
```
Comment (holgerroth) [inline nvflare/app_common/executors/statistics/statistics_executor.py:232]: There should be a check to ensure that  0.0 <= max_bins_percent <= 1.0

**Review by holgerroth — CHANGES_REQUESTED** [review]
Additional check is needed. See comment.

**Review by holgerroth — DISMISSED** [review]
LGTM

File: nvflare/app_common/executors/statistics/statistics_executor.py [hunk 2]
```diff
@@ -97,12 +124,11 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
...
     def initialize(self, fl_ctx: FLContext):
         try:
+            self.validate_inputs(fl_ctx)
             engine = fl_ctx.get_engine()
             self.stats_generator = engine.get_component(self.generator_id)
             if not isinstance(self.stats_generator, Statistics):
-                raise TypeError(
-                    f"statistics generator must implement Statistics type. Got: {type(self.stats_generator)}"
-                )
+                raise TypeError(f"{__file__} must implement `Statistics` type. Got: {type(self.stats_generator)}")
```
Comment (YuanTingHsieh) [inline nvflare/app_common/executors/statistics/statistics_executor.py:131]: Should not be ``__file__``, your file name here is statistics_executor.

**Review by YuanTingHsieh — COMMENTED** [review]
LGTM, only one minor change needed

**Review by yhwen — COMMENTED** [review]
Several places in StatisticsExecutor use print() for message logging. Those should be changed to use logger.

**Pull request closed without merging**